### PR TITLE
Google BigQuery Integration Improvements

### PR DESCRIPTION
**Pull request by MinuraPunchihewa**
## Description

This PR improves the Google BigQuery handler in order to make it Enterprise-ready.

Fixes https://github.com/mindsdb/mindsdb_private/issues/589

## Type of change

- [X] 🐛 Bug fix (non-breaking change which fixes an issue)

## Verification Process

To ensure the changes are working as expected:

 - [X]   Test Location: `tests/unit/handlers/test_bigquery.py` for unit tests and `mindsdb/integrations/handlers/bigquery_handler/tests/test_bigquery_handler.py` for integration tests.
 - [X]   Verification Steps: Run the above test modules (for integration tests, a Snowflake account will be required).

- [X] My code follows the style guidelines(PEP 8) of MindsDB.
- [X] I have appropriately commented on my code, especially in complex areas.
- [X] Necessary documentation updates are either made or tracked in issues.
- [X] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by MinuraPunchihewa** [comment]
@ZoranPandovski This is ready for the most part; I just need to run the integrations tests to make sure they work; waiting on some credentials to make that happen.
There were a couple of changes I wanted to make, but I thought I would check with you first, because they will have an impact on the front-end as well (the form mostly),
1. Add a new parameter to allow the service account key to be passed in as a URL.
2. Update the names of the existing keys from `service_account_keys` and `service_account_json` to `service_account_key_file` and `service_account_key_json` respectively. If the above is added, it would then be `service_account_key_url`.

**Comment by MinuraPunchihewa** [comment]
@ZoranPandovski This is ready for review. Please let me know what you think of the above.

File: mindsdb/integrations/handlers/bigquery_handler/bigquery_handler.py [hunk 1]
```diff
@@ -119,26 +133,49 @@ def native_query(self, query: str) -> Response:
...
     def get_tables(self) -> Response:
         """
-        Get a list with all of the tabels in BigQuery
+        Retrieves a list of all non-system tables and views in the configured dataset of the BigQuery warehouse.
+
+        Returns:
+            Response: A response object containing the list of tables and views, formatted as per the `Response` class.
+        """
+        query = f"""
+            SELECT table_name, table_schema, table_type,
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/bigquery_handler/bigquery_handler.py:156]: This shouldn't be an issue but better to remove the comma after `table_type`

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Very minor change required before we merge

File: mindsdb/integrations/handlers/bigquery_handler/bigquery_handler.py (hunk 1 above)
Comment (MinuraPunchihewa) [reply mindsdb/integrations/handlers/bigquery_handler/bigquery_handler.py]: Done.

**Comment by MinuraPunchihewa** [comment]
@ZoranPandovski I have made the requested change. Do you have any thoughts on the comments I've left above?

**Comment by ZoranPandovski** [comment]
@MinuraPunchihewa Let's move them in a separate `enhancement` issue, for now we can merge this

**Pull request merged**